### PR TITLE
Cleanup animator tracing

### PR DESCRIPTION
**Pull request by dnfield**
- Removes the `VsyncSchedulingOverhead` event, since that event didn't always make sense - it records long scary bars in frames where the schedule frame did not originate from the framework, and it's not clear to me how this is useful to our users - see https://github.com/flutter/engine/pull/20229
- Only emit `Frame Request pending` if we're going to call `BeginFrame`.
- Emit an event when we re-use the last layer tree.

Fixes https://github.com/flutter/flutter/issues/113957

### CODE REVIEW

**Comment by dnfield** [comment]
I could _probably_ write a test for this if we think this is something worth testing.

**Comment by zanderso** [comment]
Will this affect DevTools? @kenzieschmoll

File: shell/common/animator.cc [hunk 1]
```diff
@@ -208,6 +208,12 @@ void Animator::DrawLastLayerTree(
 
 void Animator::RequestFrame(bool regenerate_layer_tree) {
   if (regenerate_layer_tree) {
+    // This even will be closed by BeginFrame. BeginFrame will only be called if
```
Comment (jonahwilliams) [inline shell/common/animator.cc:211]:
```suggestion
    // This event will be closed by BeginFrame. BeginFrame will only be called if
```

**Review by zanderso — APPROVED** [review]
I'm not seeing the string `VsyncSchedulingOverhead` in the framework tree anywhere, so if DevTools isn't using it, I think it's safe to remove.

**Review by chinmaygarde — APPROVED** [review]

**Comment by kenzieschmoll** [comment]
LGTM from DevTools perspective. This change will not affect our tracing logic. Thank you for the tag @zanderso.

**Comment by zanderso** [comment]
The formatter wants the comment whitespace to be fixed.

**Pull request merged**